addons/mail/models/mail_message.py

```diff
@@ -855,6 +857,21 @@ def _message_format(self, fnames, format_reply=True, legacy=False):
                 'id': message_sudo.author_guest_id.id,
                 'name': message_sudo.author_guest_id.name,
             } if message_sudo.author_guest_id else [('clear',)]
+            if message_sudo.author_guest_id:
+                sender = guestAuthor
```

Comment:
JS needs to know if the sender is a guest or a partner, mixing the id/name object of both will lead to consistency issue (replacing partner name matching id by guest name).
Guest needs a separate key.

Reply:
Yes it's fine to ignore the special display in JS if guest author is set.